common/lcs/src/ser.rs

```diff
@@ -419,15 +418,17 @@ impl<'a> ser::SerializeMap for MapSerializer<'a> {
         Ok(())
     }
 
-    fn end(self) -> Result<()> {
+    fn end(mut self) -> Result<()> {
         if self.next_key.is_some() {
             return Err(Error::ExpectedMapValue);
         }
+        self.entries.sort_by(|e1, e2| e1.0.cmp(&e2.0));
+        // Not checking for duplicate keys: unicity must be enforced by the Rust map.
```

Comment:
Welp, you've found another potential issue. Do we care that we weren't (and still aren't) checking for unicity of keys? In practice we shouldn't have an issue if we're using the derive macros, but this could be broken if someone decided they wanted to hand implement the `Serialize` trait and send multiples of the same key?

Also i'm assuming we do check for unicity on the de side correct?

Reply:
Well, we do (now) check for unicity on the deserialization side.

Surely, we could call [Vec::dedup](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup) and pay another pass on the data, but is it worth it?

IMHO, this sorting code is really just to make `HashMap`s work as intended. (`BTreeMap`s don't need anything at all.)

Reply:
Fair enough, and idk if in practice if we'll actually hit this. At the very lease we have a comment indicating the behavior

Reply:
I'm just going to add a call `dedup_by` because the difference is very small anyway.